zerver/decorator.py

```diff
@@ -250,7 +250,7 @@ def decorator(view_func):
         @wraps(view_func, assigned=available_attrs(view_func))
         def _wrapped_view(request, *args, **kwargs):
             # type: (HttpRequest, *Any, **Any) -> HttpResponse
-            if test_func(request.user):
+            if test_func(request):
```

Comment:
Can you move the API change here to pass an `HttpRequest` rather than a UserProfile into its own commit at the start of the branch?  Then the main commit can just add the `REALMS_HAVE_SUBDOMAINS` code to this.